# OLYMPIC DATASET
Dataset

NOC: https://drive.google.com/file/d/1_zGhADzXEA9mFHvuCvpllXMCN2DUE2tH/view?usp=sharing

Athlete: https://drive.google.com/file/d/1y6e0eNKfyo9ILqQHszcrRuv3uQhIMm_z/view?usp=sharing

In [142]:
# we need pandas for datamanupilation and cleaning
import pandas as pd

# we need numpy for acsessing array in dataset
import numpy as np

# we need re (regex) in case of data cleaning
import re 

# we need matplotlib , tablely and seaborn  for data vizualisation
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as pl

* download the dataset store it in your local
* use the file path to get the data in as a dataframe

In [143]:
# creating a dataframe using read_csv
ath = pd.read_csv(r"C:\Users\shoba\Desktop\data_science\olympic_dataset\athlete_events.csv")
noc = pd.read_csv(r'C:\Users\shoba\Desktop\data_science\olympic_dataset\noc_regions.csv')

In [144]:
ath

ID                      Name Gender   Age  Height  Weight  \
0            1                 A Dijiang      M  24.0   180.0    80.0   
1            2                  A Lamusi      M  23.0   170.0    60.0   
2            3       Gunnar Nielsen Aaby      M  24.0     NaN     NaN   
3            4      Edgar Lindenau Aabye      M  34.0     NaN     NaN   
4            5  Christine Jacoba Aaftink      F  21.0   185.0    82.0   
...        ...                       ...    ...   ...     ...     ...   
271111  135569                Andrzej ya      M  29.0   179.0    89.0   
271112  135570                  Piotr ya      M  27.0   176.0    59.0   
271113  135570                  Piotr ya      M  27.0   176.0    59.0   
271114  135571        Tomasz Ireneusz ya      M  30.0   185.0    96.0   
271115  135571        Tomasz Ireneusz ya      M  34.0   185.0    96.0   

                  Team  NOC        Games  Year  Season            City  \
0                China  CHN  1992 Summer  1992  Summer       Barcelona   
1                China  CHN  2012 Summer  2012  Summer          London   
2              Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer           Paris   
4          Netherlands  NED  1988 Winter  1988  Winter         Calgary   
...                ...  ...          ...   ...     ...             ...   
271111        Poland-1  POL  1976 Winter  1976  Winter       Innsbruck   
271112          Poland  POL  2014 Winter  2014  Winter           Sochi   
271113          Poland  POL  2014 Winter  2014  Winter           Sochi   
271114          Poland  POL  1998 Winter  1998  Winter          Nagano   
271115          Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

                Sport                                     Event Medal  
0          Basketball               Basketball Men's Basketball   NaN  
1                Judo              Judo Men's Extra-Lightweight   NaN  
2            Football                   Football Men's Football   NaN  
3          Tug-Of-War               Tug-Of-War Men's Tug-Of-War  Gold  
4       Speed Skating          Speed Skating Women's 500 metres   NaN  
...               ...                                       ...   ...  
271111           Luge                Luge Mixed (Men)'s Doubles   NaN  
271112    Ski Jumping  Ski Jumping Men's Large Hill, Individual   NaN  
271113    Ski Jumping        Ski Jumping Men's Large Hill, Team   NaN  
271114      Bobsleigh                      Bobsleigh Men's Four   NaN  
271115      Bobsleigh                      Bobsleigh Men's Four   NaN  

[271116 rows x 15 columns]

In [145]:
noc

NOC       region                 notes
0    AFG  Afghanistan                   NaN
1    AHO      Curacao  Netherlands Antilles
2    ALB      Albania                   NaN
3    ALG      Algeria                   NaN
4    AND      Andorra                   NaN
..   ...          ...                   ...
225  YEM        Yemen                   NaN
226  YMD        Yemen           South Yemen
227  YUG       Serbia            Yugoslavia
228  ZAM       Zambia                   NaN
229  ZIM     Zimbabwe                   NaN

[230 rows x 3 columns]

* we can see that the noc df has less column 
* so we can start the datacleaning from the nac df

In [146]:
noc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 230 entries, 0 to 229
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   NOC     230 non-null    object
 1   region  227 non-null    object
 2   notes   21 non-null     object
dtypes: object(3)
memory usage: 5.5+ KB


* noc df has 3 columns and 230 rows
* NOC column contains no null value
    * still we have too see any irrelevent data present's in that
* region column contains sum null value 
    * so we can use manupulation method's to rectify the nan value
* notes column has some values only and more null value
    * we cannot use manupulation as there are more null values
    * and it's seem's that we don't have relatable data in that 
    * so we can drop the entire notes column and it will not affect the data

In [147]:
noc.NOC.unique()

array(['AFG', 'AHO', 'ALB', 'ALG', 'AND', 'ANG', 'ANT', 'ANZ', 'ARG',
       'ARM', 'ARU', 'ASA', 'AUS', 'AUT', 'AZE', 'BAH', 'BAN', 'BAR',
       'BDI', 'BEL', 'BEN', 'BER', 'BHU', 'BIH', 'BIZ', 'BLR', 'BOH',
       'BOL', 'BOT', 'BRA', 'BRN', 'BRU', 'BUL', 'BUR', 'CAF', 'CAM',
       'CAN', 'CAY', 'CGO', 'CHA', 'CHI', 'CHN', 'CIV', 'CMR', 'COD',
       'COK', 'COL', 'COM', 'CPV', 'CRC', 'CRO', 'CRT', 'CUB', 'CYP',
       'CZE', 'DEN', 'DJI', 'DMA', 'DOM', 'ECU', 'EGY', 'ERI', 'ESA',
       'ESP', 'EST', 'ETH', 'EUN', 'FIJ', 'FIN', 'FRA', 'FRG', 'FSM',
       'GAB', 'GAM', 'GBR', 'GBS', 'GDR', 'GEO', 'GEQ', 'GER', 'GHA',
       'GRE', 'GRN', 'GUA', 'GUI', 'GUM', 'GUY', 'HAI', 'HKG', 'HON',
       'HUN', 'INA', 'IND', 'IOA', 'IRI', 'IRL', 'IRQ', 'ISL', 'ISR',
       'ISV', 'ITA', 'IVB', 'JAM', 'JOR', 'JPN', 'KAZ', 'KEN', 'KGZ',
       'KIR', 'KOR', 'KOS', 'KSA', 'KUW', 'LAO', 'LAT', 'LBA', 'LBR',
       'LCA', 'LES', 'LIB', 'LIE', 'LTU', 'LUX', 'MAD', 'MAL', 'MAR',
       'MAS', 'MAW',

* we can conclude that there is no null as well as no irrelevent data 
* we can find that it contains country code

In [148]:
noc.region.isna().sum()

3

In [149]:
noc.region.value_counts()

Germany           4
Yemen             3
Czech Republic    3
Serbia            3
Russia            3
                 ..
Ghana             1
Grenada           1
Guatemala         1
Guinea            1
Zambia            1
Name: region, Length: 206, dtype: int64

In [150]:
noc[noc.region.isna()]

NOC region                 notes
168  ROT    NaN  Refugee Olympic Team
208  TUV    NaN                Tuvalu
213  UNK    NaN               Unknown

* we can see that 
    * ROT means refugee olympic team
    * TUV means tuvalu
    * UNK means unknown
* instead of having null value we can have the notes information over there 

In [151]:
noc['region'][168] = noc['notes'][168]
noc['region'][208] = noc['notes'][208]
noc['region'][213] = noc['notes'][213]

In [152]:
noc.region.isna().sum()

0

* so we have cleared the null values in the region column

In [153]:
noc.notes.isnull().sum()

209

* we can see here out of 230 datapoint 209 have null values so we are droping the notes column

In [154]:
# droping the notes column
noc.drop("notes", axis=1, inplace=True)

In [155]:
noc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 230 entries, 0 to 229
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   NOC     230 non-null    object
 1   region  230 non-null    object
dtypes: object(2)
memory usage: 3.7+ KB


In [156]:
noc.duplicated().sum()

0

* we can see that the dataframe is clean and neat.
* and there are no duplicated value over there

In [157]:
ath.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 15 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   ID      271116 non-null  int64  
 1   Name    271116 non-null  object 
 2   Gender  271116 non-null  object 
 3   Age     261642 non-null  float64
 4   Height  210945 non-null  float64
 5   Weight  208241 non-null  float64
 6   Team    271116 non-null  object 
 7   NOC     271116 non-null  object 
 8   Games   271116 non-null  object 
 9   Year    271116 non-null  int64  
 10  Season  271116 non-null  object 
 11  City    271116 non-null  object 
 12  Sport   271116 non-null  object 
 13  Event   271116 non-null  object 
 14  Medal   39783 non-null   object 
dtypes: float64(3), int64(2), object(10)
memory usage: 31.0+ MB


* we can see that the ath dataframe contain's 15 columns and 271116 datpoints 
    * column's -> ID, Name, Gender, Team, NOC, Games, Year, Season, City, Sport, Event
        * these columns have no null values 
        * we still have to look for inapropriate values and duplicate values in these column's
        * and we can see that NOC column presents in both data set 
        * we combine the both dataset using the NOC values
    * column's -> Age, Height, Weight, medal
        * these columns contains null values
        * we have to use data manupulation to get rid of null value
        * we can drop the column if the data is not needed 

In [158]:
ath.columns

Index(['ID', 'Name', 'Gender', 'Age', 'Height', 'Weight', 'Team', 'NOC',
       'Games', 'Year', 'Season', 'City', 'Sport', 'Event', 'Medal'],
      dtype='object')

In [159]:
ath.ID

0              1
1              2
2              3
3              4
4              5
           ...  
271111    135569
271112    135570
271113    135570
271114    135571
271115    135571
Name: ID, Length: 271116, dtype: int64

* this column contains the index of each person 
* we can see that the number 135570 is repeated its because the same person participated in the different games
* we don't need this data that much
* and it not contain's any null value

In [160]:
ath.Name.value_counts()

Robert Tait McKenzie            58
Heikki Ilmari Savolainen        39
Joseph "Josy" Stoffel           38
Ioannis Theofilakis             36
Takashi Ono                     33
                                ..
Tatyana Vasilyevna Kalmykova     1
Mariya Lvovna Kalmykova          1
Christine Kalmer                 1
Joannis "Jannis" Kalmazidis      1
Pierre-Georges LeClercq          1
Name: Name, Length: 134732, dtype: int64

* we can see that mckenzie has occured 58 times 

In [161]:
ath[ath.Name.duplicated()]

ID                          Name Gender   Age  Height  Weight  \
5            5      Christine Jacoba Aaftink      F  21.0   185.0    82.0   
6            5      Christine Jacoba Aaftink      F  25.0   185.0    82.0   
7            5      Christine Jacoba Aaftink      F  25.0   185.0    82.0   
8            5      Christine Jacoba Aaftink      F  27.0   185.0    82.0   
9            5      Christine Jacoba Aaftink      F  27.0   185.0    82.0   
...        ...                           ...    ...   ...     ...     ...   
271103  135563     Olesya Nikolayevna Zykina      F  23.0   171.0    64.0   
271106  135565      Fernando scar Zylberberg      M  27.0   168.0    76.0   
271109  135567  Aleksandr Viktorovich Zyuzin      M  28.0   183.0    72.0   
271113  135570                      Piotr ya      M  27.0   176.0    59.0   
271115  135571            Tomasz Ireneusz ya      M  34.0   185.0    96.0   

               Team  NOC        Games  Year  Season            City  \
5       Netherlands  NED  1988 Winter  1988  Winter         Calgary   
6       Netherlands  NED  1992 Winter  1992  Winter     Albertville   
7       Netherlands  NED  1992 Winter  1992  Winter     Albertville   
8       Netherlands  NED  1994 Winter  1994  Winter     Lillehammer   
9       Netherlands  NED  1994 Winter  1994  Winter     Lillehammer   
...             ...  ...          ...   ...     ...             ...   
271103       Russia  RUS  2004 Summer  2004  Summer          Athina   
271106    Argentina  ARG  2004 Summer  2004  Summer          Athina   
271109       Russia  RUS  2004 Summer  2004  Summer          Athina   
271113       Poland  POL  2014 Winter  2014  Winter           Sochi   
271115       Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

                Sport                                   Event   Medal  
5       Speed Skating      Speed Skating Women's 1,000 metres     NaN  
6       Speed Skating        Speed Skating Women's 500 metres     NaN  
7       Speed Skating      Speed Skating Women's 1,000 metres     NaN  
8       Speed Skating        Speed Skating Women's 500 metres     NaN  
9       Speed Skating      Speed Skating Women's 1,000 metres     NaN  
...               ...                                     ...     ...  
271103      Athletics  Athletics Women's 4 x 400 metres Relay  Silver  
271106         Hockey                     Hockey Men's Hockey     NaN  
271109         Rowing  Rowing Men's Lightweight Coxless Fours     NaN  
271113    Ski Jumping      Ski Jumping Men's Large Hill, Team     NaN  
271115      Bobsleigh                    Bobsleigh Men's Four     NaN  

[136384 rows x 15 columns]

* we can say that the duplicated values have different kind of data for different columns

In [162]:
ath.Gender.value_counts()

M    196594
F     74522
Name: Gender, dtype: int64

* we can see that this column has only 2 values m -male, f - female
    * instead of having m and f as value's we can convert that to 0 and 1 respectively
    * this column has have no null value and also has no in appropriate values

In [163]:
# converting the m and f into 0 and 1 
ath.Gender.replace(['M','F'], [0,1], inplace=True)

In [164]:
ath.Gender.value_counts()

0    196594
1     74522
Name: Gender, dtype: int64

* we have converted the values of the m and f to 0 and 1 so that we can use this data for analysis

In [165]:
ath.Age.isnull().sum()

9474

In [166]:
ath.Age.describe()

count    261642.000000
mean         25.556898
std           6.393561
min          10.000000
25%          21.000000
50%          24.000000
75%          28.000000
max          97.000000
Name: Age, dtype: float64

* we have mean = 25
* we have median  = 24
* we can choose either one of them to fill the nan value

In [167]:
ath[ath.Age.isnull()]

ID                   Name  Gender  Age  Height  Weight  \
147         54  Mohamed Jamshid Abadi       0  NaN     NaN     NaN   
152         58       Georgi Abadzhiev       0  NaN     NaN     NaN   
153         58       Georgi Abadzhiev       0  NaN     NaN     NaN   
162         66        Mohamed Abakkar       0  NaN   156.0    48.0   
212        102      Sayed Fahmy Abaza       0  NaN     NaN     NaN   
...        ...                    ...     ...  ...     ...     ...   
270679  135368      Antonio A. Zucchi       0  NaN     NaN     NaN   
270774  135416              Star Zulu       0  NaN     NaN     NaN   
270793  135426           Max Zumstein       0  NaN     NaN     NaN   
270804  135434     Edmundo Ziga Erraz       0  NaN   180.0    95.0   
270805  135435             Faelo Ziga       0  NaN     NaN     NaN   

               Team  NOC        Games  Year  Season         City      Sport  \
147            Iran  IRI  1948 Summer  1948  Summer       London     Boxing   
152        Bulgaria  BUL  1924 Summer  1924  Summer        Paris    Cycling   
153        Bulgaria  BUL  1924 Summer  1924  Summer        Paris    Cycling   
162           Sudan  SUD  1972 Summer  1972  Summer       Munich     Boxing   
212           Egypt  EGY  1920 Summer  1920  Summer    Antwerpen   Football   
...             ...  ...          ...   ...     ...          ...        ...   
270679    Argentina  ARG  1948 Summer  1948  Summer       London     Hockey   
270774       Zambia  ZAM  1984 Summer  1984  Summer  Los Angeles     Boxing   
270793  Switzerland  SUI  1928 Summer  1928  Summer    Amsterdam     Hockey   
270804        Chile  CHI  1948 Summer  1948  Summer       London  Athletics   
270805        Chile  CHI  1928 Summer  1928  Summer    Amsterdam   Swimming   

                                      Event Medal  
147                Boxing Men's Heavyweight   NaN  
152     Cycling Men's Road Race, Individual   NaN  
153           Cycling Men's Road Race, Team   NaN  
162                  Boxing Men's Flyweight   NaN  
212                 Football Men's Football   NaN  
...                                     ...   ...  
270679                  Hockey Men's Hockey   NaN  
270774            Boxing Men's Bantamweight   NaN  
270793                  Hockey Men's Hockey   NaN  
270804         Athletics Men's Hammer Throw   NaN  
270805  Swimming Men's 100 metres Freestyle   NaN  

[9474 rows x 15 columns]

* we have 9475 null values in the age column 
* and we are going to fill the null value with the mean of age column mean value  = 25
* the dtype of the age column is float and we have convert that into a int

In [168]:
# replacing nan values with the mean 
ath.Age.fillna(ath.Age.mean(),inplace=True) 

In [169]:
ath.Age.isna().sum()

0

* we have filled the nan values with mean.

In [170]:
ath.Age.dtype

dtype('float64')

* we can see that dtype is float 
* we have convert it into a int

In [171]:
ath['Age'] = ath['Age'].astype(np.int64)

In [172]:
ath.Age.dtype

dtype('int64')

* we have converted the dtype to the int64

In [173]:
ath.Height

0         180.0
1         170.0
2           NaN
3           NaN
4         185.0
          ...  
271111    179.0
271112    176.0
271113    176.0
271114    185.0
271115    185.0
Name: Height, Length: 271116, dtype: float64

In [174]:
ath.Height.describe()

count    210945.000000
mean        175.338970
std          10.518462
min         127.000000
25%         168.000000
50%         175.000000
75%         183.000000
max         226.000000
Name: Height, dtype: float64

In [175]:
ath.Height.isnull().sum()

60171

* we have 60171 null values in the height column
* we have to fill the null values with the mean (mean =  175)
* we have to change dtype to int

In [176]:
# filling the height column with the mean value
ath.Height.fillna(ath.Height.mean(),inplace=True) 

In [177]:
ath.Height.isnull().sum()

0

In [178]:
ath['Height'] = ath['Height'].astype(np.int64)

* filled null values with the mean
* and converted the dtype to int

In [179]:
ath.Weight.isna().sum()

62875

In [180]:
ath.Weight.describe()

count    208241.000000
mean         70.702393
std          14.348020
min          25.000000
25%          60.000000
50%          70.000000
75%          79.000000
max         214.000000
Name: Weight, dtype: float64

* in the weight column we have 62875 null values
* and we can use the mean here as well to fill the nan
* we have to convert the dtype to int 

In [181]:
# fill nan with the mean 
ath.Weight.fillna(ath.Weight.mean(), inplace=True)

In [182]:
# converting the dtype to int
ath['Weight'] = ath['Weight'].astype(np.int64)

In [183]:
ath.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 15 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   ID      271116 non-null  int64 
 1   Name    271116 non-null  object
 2   Gender  271116 non-null  int64 
 3   Age     271116 non-null  int64 
 4   Height  271116 non-null  int64 
 5   Weight  271116 non-null  int64 
 6   Team    271116 non-null  object
 7   NOC     271116 non-null  object
 8   Games   271116 non-null  object
 9   Year    271116 non-null  int64 
 10  Season  271116 non-null  object
 11  City    271116 non-null  object
 12  Sport   271116 non-null  object
 13  Event   271116 non-null  object
 14  Medal   39783 non-null   object
dtypes: int64(6), object(9)
memory usage: 31.0+ MB


* we have filled the nan value and converted the dtype to int

In [184]:
ath.Team

0                  China
1                  China
2                Denmark
3         Denmark/Sweden
4            Netherlands
               ...      
271111          Poland-1
271112            Poland
271113            Poland
271114            Poland
271115            Poland
Name: Team, Length: 271116, dtype: object

In [185]:
ath.Team.value_counts()

United States    17847
France           11988
Great Britain    11404
Italy            10260
Germany           9326
                 ...  
Briar                1
Hannover             1
Nan-2                1
Brentina             1
Digby                1
Name: Team, Length: 1184, dtype: int64

In [186]:
ath.Team.isna().sum()

0

In [187]:
ath.Team.nunique()

1184

In [188]:
ath.Team.unique()

array(['China', 'Denmark', 'Denmark/Sweden', ..., 'Solos Carex',
       'Dow Jones', 'Digby'], dtype=object)

* column Team contains no null values
* it contains about 1184 unique values 

In [189]:
ath.NOC.unique()

array(['CHN', 'DEN', 'NED', 'USA', 'FIN', 'NOR', 'ROU', 'EST', 'FRA',
       'MAR', 'ESP', 'EGY', 'IRI', 'BUL', 'ITA', 'CHA', 'AZE', 'SUD',
       'RUS', 'ARG', 'CUB', 'BLR', 'GRE', 'CMR', 'TUR', 'CHI', 'MEX',
       'URS', 'NCA', 'HUN', 'NGR', 'ALG', 'KUW', 'BRN', 'PAK', 'IRQ',
       'UAR', 'LIB', 'QAT', 'MAS', 'GER', 'CAN', 'IRL', 'AUS', 'RSA',
       'ERI', 'TAN', 'JOR', 'TUN', 'LBA', 'BEL', 'DJI', 'PLE', 'COM',
       'KAZ', 'BRU', 'IND', 'KSA', 'SYR', 'MDV', 'ETH', 'UAE', 'YAR',
       'INA', 'PHI', 'SGP', 'UZB', 'KGZ', 'TJK', 'EUN', 'JPN', 'CGO',
       'SUI', 'BRA', 'FRG', 'GDR', 'MON', 'ISR', 'URU', 'SWE', 'ISV',
       'SRI', 'ARM', 'CIV', 'KEN', 'BEN', 'UKR', 'GBR', 'GHA', 'SOM',
       'LAT', 'NIG', 'MLI', 'AFG', 'POL', 'CRC', 'PAN', 'GEO', 'SLO',
       'CRO', 'GUY', 'NZL', 'POR', 'PAR', 'ANG', 'VEN', 'COL', 'BAN',
       'PER', 'ESA', 'PUR', 'UGA', 'HON', 'ECU', 'TKM', 'MRI', 'SEY',
       'TCH', 'LUX', 'MTN', 'CZE', 'SKN', 'TTO', 'DOM', 'VIN', 'JAM',
       'LBR', 'SUR',

In [190]:
ath.NOC.nunique()

230

* we can clearly see that the NOC column reacts as a bridge between the two dataframe 
* we can use this column to merge the region column in the ath dataframe 
* after there is no use of this column so we can drop this one

In [191]:
# merging the region column as the 
region = ath['NOC'].map(noc.set_index('NOC')['region']) # creating a variable containing the region data as per the noc in the ath dataframe
ath.insert(7, 'region', region) # creating a new column in the dataframe at the 7th position as "region"

In [205]:
# dropping the NOC column 
ath.drop("NOC",axis=1,inplace=True)

In [193]:
ath.head(10)

ID                      Name  Gender  Age  Height  Weight            Team  \
0   1                 A Dijiang       0   24     180      80           China   
1   2                  A Lamusi       0   23     170      60           China   
2   3       Gunnar Nielsen Aaby       0   24     175      70         Denmark   
3   4      Edgar Lindenau Aabye       0   34     175      70  Denmark/Sweden   
4   5  Christine Jacoba Aaftink       1   21     185      82     Netherlands   
5   5  Christine Jacoba Aaftink       1   21     185      82     Netherlands   
6   5  Christine Jacoba Aaftink       1   25     185      82     Netherlands   
7   5  Christine Jacoba Aaftink       1   25     185      82     Netherlands   
8   5  Christine Jacoba Aaftink       1   27     185      82     Netherlands   
9   5  Christine Jacoba Aaftink       1   27     185      82     Netherlands   

        region  NOC        Games  Year  Season         City          Sport  \
0        China  CHN  1992 Summer  1992  Summer    Barcelona     Basketball   
1        China  CHN  2012 Summer  2012  Summer       London           Judo   
2      Denmark  DEN  1920 Summer  1920  Summer    Antwerpen       Football   
3      Denmark  DEN  1900 Summer  1900  Summer        Paris     Tug-Of-War   
4  Netherlands  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
5  Netherlands  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
6  Netherlands  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
7  Netherlands  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
8  Netherlands  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   
9  Netherlands  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   

                                Event Medal  
0         Basketball Men's Basketball   NaN  
1        Judo Men's Extra-Lightweight   NaN  
2             Football Men's Football   NaN  
3         Tug-Of-War Men's Tug-Of-War  Gold  
4    Speed Skating Women's 500 metres   NaN  
5  Speed Skating Women's 1,000 metres   NaN  
6    Speed Skating Women's 500 metres   NaN  
7  Speed Skating Women's 1,000 metres   NaN  
8    Speed Skating Women's 500 metres   NaN  
9  Speed Skating Women's 1,000 metres   NaN

* we can see here that the games column the same data in the column (year and season)
* so we can drop the column 

In [194]:
# droping the Games column from the dataframe
ath.drop("Games", axis=1, inplace=True)

In [195]:
ath.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 15 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   ID      271116 non-null  int64 
 1   Name    271116 non-null  object
 2   Gender  271116 non-null  int64 
 3   Age     271116 non-null  int64 
 4   Height  271116 non-null  int64 
 5   Weight  271116 non-null  int64 
 6   Team    271116 non-null  object
 7   region  270767 non-null  object
 8   NOC     271116 non-null  object
 9   Year    271116 non-null  int64 
 10  Season  271116 non-null  object
 11  City    271116 non-null  object
 12  Sport   271116 non-null  object
 13  Event   271116 non-null  object
 14  Medal   39783 non-null   object
dtypes: int64(6), object(9)
memory usage: 31.0+ MB


In [196]:
ath.region.value_counts()

USA            18853
Germany        15883
France         12758
UK             12256
Russia         11692
               ...  
Timor-Leste        9
Kosovo             8
Tuvalu             7
South Sudan        3
Unknown            2
Name: region, Length: 208, dtype: int64

In [197]:
ath.region.isnull().sum()

349

In [198]:
ath[ath.region.isnull()]

ID                             Name  Gender  Age  Height  Weight  \
578        332                 Zarinah Abdullah       1   21     175      70   
579        332                 Zarinah Abdullah       1   25     175      70   
2719      1515  Saiyidah Aisyah Mohamed Rafa'ee       1   28     173      68   
7659      4264                   Ang Peng Siong       0   21     180      75   
7660      4264                   Ang Peng Siong       0   21     180      75   
...        ...                              ...     ...  ...     ...     ...   
268752  134454                    Zhang Guirong       1   26     182      95   
268753  134454                    Zhang Guirong       1   30     182      95   
268993  134587                    Zhang Xueling       1   21     163      53   
268994  134587                    Zhang Xueling       1   21     163      53   
269455  134826                       Zhou Yihan       1   22     168      54   

               Team region  NOC  Year  Season            City         Sport  \
578       Singapore    NaN  SGP  1992  Summer       Barcelona     Badminton   
579       Singapore    NaN  SGP  1996  Summer         Atlanta     Badminton   
2719      Singapore    NaN  SGP  2016  Summer  Rio de Janeiro        Rowing   
7659      Singapore    NaN  SGP  1984  Summer     Los Angeles      Swimming   
7660      Singapore    NaN  SGP  1984  Summer     Los Angeles      Swimming   
...             ...    ...  ...   ...     ...             ...           ...   
268752    Singapore    NaN  SGP  2004  Summer          Athina     Athletics   
268753    Singapore    NaN  SGP  2008  Summer         Beijing     Athletics   
268993    Singapore    NaN  SGP  2004  Summer          Athina  Table Tennis   
268994  Singapore-1    NaN  SGP  2004  Summer          Athina  Table Tennis   
269455    Singapore    NaN  SGP  2016  Summer  Rio de Janeiro  Table Tennis   

                                                Event Medal  
578                         Badminton Women's Singles   NaN  
579                         Badminton Women's Singles   NaN  
2719                     Rowing Women's Single Sculls   NaN  
7659              Swimming Men's 100 metres Freestyle   NaN  
7660    Swimming Men's 4 x 100 metres Freestyle Relay   NaN  
...                                               ...   ...  
268752                     Athletics Women's Shot Put   NaN  
268753                     Athletics Women's Shot Put   NaN  
268993                   Table Tennis Women's Singles   NaN  
268994                   Table Tennis Women's Doubles   NaN  
269455                      Table Tennis Women's Team   NaN  

[349 rows x 15 columns]

* its seems only the singapore part got null values 
* its the colum merged from the NOC dataframe 
* we would check if there is singupore in it 

In [203]:
noc[noc.region == "Singapore"]

NOC     region
178  SIN  Singapore

* we can see that the NOC code for singapore is diiferent in the ath data frame 
* so we can simply replace the nan with singapore 

In [222]:
ath.region.fillna("Singapore", inplace=True)

* this column has 349 null values

In [206]:
ath.head(2)

ID       Name  Gender  Age  Height  Weight   Team region  Year  Season  \
0   1  A Dijiang       0   24     180      80  China  China  1992  Summer   
1   2   A Lamusi       0   23     170      60  China  China  2012  Summer   

        City       Sport                         Event Medal  
0  Barcelona  Basketball   Basketball Men's Basketball   NaN  
1     London        Judo  Judo Men's Extra-Lightweight   NaN

In [207]:
ath.Year.nunique()

35

In [208]:
ath.Year.value_counts()

1992    16413
1988    14676
2000    13821
1996    13780
2016    13688
2008    13602
2004    13443
2012    12920
1972    11959
1984    11588
1976    10502
1968    10479
1964     9480
1952     9358
1960     9235
1980     8937
1948     7480
1936     7401
1956     6434
1924     5693
1928     5574
2014     4891
2010     4402
2006     4382
1920     4292
2002     4109
1912     4040
1998     3605
1932     3321
1994     3160
1908     3101
1900     1936
1906     1733
1904     1301
1896      380
Name: Year, dtype: int64

In [210]:
ath.Year.isnull().sum()

0

* there is no null values in this column 
* and the dtype is int  

In [211]:
ath.Season.isnull().sum()

0

In [212]:
ath.Season.nunique()

2

In [213]:
ath.Season.unique()

array(['Summer', 'Winter'], dtype=object)

* instead of having summer and winter we can have o and 1
* replace o and 1 for summer and winter respectively so that we do analysis on it 

In [215]:
# replacing summer with 0 and winter with 1
ath.Season.replace(["Summer", 'Winter'], [0, 1], inplace=True)

In [217]:
ath.Season.value_counts()

0    222552
1     48564
Name: Season, dtype: int64

In [224]:
ath.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 14 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   ID      271116 non-null  int64 
 1   Name    271116 non-null  object
 2   Gender  271116 non-null  int64 
 3   Age     271116 non-null  int64 
 4   Height  271116 non-null  int64 
 5   Weight  271116 non-null  int64 
 6   Team    271116 non-null  object
 7   region  271116 non-null  object
 8   Year    271116 non-null  int64 
 9   Season  271116 non-null  int64 
 10  City    271116 non-null  object
 11  Sport   271116 non-null  object
 12  Event   271116 non-null  object
 13  Medal   39783 non-null   object
dtypes: int64(7), object(7)
memory usage: 29.0+ MB


In [225]:
ath.City.nunique()

42

In [226]:
ath.City.value_counts()

London                    22426
Athina                    15556
Sydney                    13821
Atlanta                   13780
Rio de Janeiro            13688
Beijing                   13602
Barcelona                 12977
Los Angeles               12423
Seoul                     12037
Munich                    10304
Montreal                   8641
Mexico City                8588
Helsinki                   8270
Roma                       8119
Tokyo                      7702
Moskva                     7191
Paris                      7169
Berlin                     6506
Amsterdam                  4992
Sochi                      4891
Melbourne                  4829
Vancouver                  4402
Torino                     4382
Stockholm                  4338
Antwerpen                  4292
Salt Lake City             4109
Innsbruck                  3639
Nagano                     3605
Albertville                3436
Lillehammer                3160
Calgary                    2639
Sarajevo

In [227]:
ath.head()

ID                      Name  Gender  Age  Height  Weight            Team  \
0   1                 A Dijiang       0   24     180      80           China   
1   2                  A Lamusi       0   23     170      60           China   
2   3       Gunnar Nielsen Aaby       0   24     175      70         Denmark   
3   4      Edgar Lindenau Aabye       0   34     175      70  Denmark/Sweden   
4   5  Christine Jacoba Aaftink       1   21     185      82     Netherlands   

        region  Year  Season       City          Sport  \
0        China  1992       0  Barcelona     Basketball   
1        China  2012       0     London           Judo   
2      Denmark  1920       0  Antwerpen       Football   
3      Denmark  1900       0      Paris     Tug-Of-War   
4  Netherlands  1988       1    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [228]:
ath.Sport.nunique()

66

In [229]:
ath.Sport.value_counts()

Athletics        38624
Gymnastics       26707
Swimming         23195
Shooting         11448
Cycling          10859
                 ...  
Racquets            12
Jeu De Paume        11
Roque                4
Basque Pelota        2
Aeronautics          1
Name: Sport, Length: 66, dtype: int64

In [230]:
ath.Event.nunique()

765

In [231]:
ath.head(2)

ID       Name  Gender  Age  Height  Weight   Team region  Year  Season  \
0   1  A Dijiang       0   24     180      80  China  China  1992       0   
1   2   A Lamusi       0   23     170      60  China  China  2012       0   

        City       Sport                         Event Medal  
0  Barcelona  Basketball   Basketball Men's Basketball   NaN  
1     London        Judo  Judo Men's Extra-Lightweight   NaN

* we only wanted the event but it has an extra data which we have that on sport column
* we are replacing the event data without the sport name before

In [234]:
for i,j in zip(range(len(ath.Event)), ath.Sport):
    ath.Event[i] = re.sub(f"{j}\s",'',ath.Event[i])

C:\Users\shoba\AppData\Local\Temp/ipykernel_7104/3760532301.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ath.Event[i] = re.sub(f"{j}\s",'',ath.Event[i])


In [235]:
ath.head(2)

ID       Name  Gender  Age  Height  Weight   Team region  Year  Season  \
0   1  A Dijiang       0   24     180      80  China  China  1992       0   
1   2   A Lamusi       0   23     170      60  China  China  2012       0   

        City       Sport                    Event Medal  
0  Barcelona  Basketball         Men's Basketball   NaN  
1     London        Judo  Men's Extra-Lightweight   NaN

In [236]:
ath.Medal.isnull().sum()

231333

In [237]:
ath.Medal.value_counts()

Gold      13372
Bronze    13295
Silver    13116
Name: Medal, dtype: int64

* we can say that the nan values are non winning players
* instead of having gold, bronze and silver 
* we can use 0,1,2,3 for nan,bronze ,silver,gold respectively

In [239]:
# replacing the nan value with the 0
ath.Medal.fillna(0, inplace=True)

In [240]:
ath.Medal.value_counts()

0         231333
Gold       13372
Bronze     13295
Silver     13116
Name: Medal, dtype: int64

In [242]:
# replacing the gold silver bronce value
ath.Medal.replace(['Bronze', 'Silver', 'Gold'], [1, 2, 3], inplace=True)

In [243]:
ath.head()

ID                      Name  Gender  Age  Height  Weight            Team  \
0   1                 A Dijiang       0   24     180      80           China   
1   2                  A Lamusi       0   23     170      60           China   
2   3       Gunnar Nielsen Aaby       0   24     175      70         Denmark   
3   4      Edgar Lindenau Aabye       0   34     175      70  Denmark/Sweden   
4   5  Christine Jacoba Aaftink       1   21     185      82     Netherlands   

        region  Year  Season       City          Sport  \
0        China  1992       0  Barcelona     Basketball   
1        China  2012       0     London           Judo   
2      Denmark  1920       0  Antwerpen       Football   
3      Denmark  1900       0      Paris     Tug-Of-War   
4  Netherlands  1988       1    Calgary  Speed Skating   

                     Event  Medal  
0         Men's Basketball      0  
1  Men's Extra-Lightweight      0  
2           Men's Football      0  
3         Men's Tug-Of-War      3  
4       Women's 500 metres      0

In [244]:
ath.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 14 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   ID      271116 non-null  int64 
 1   Name    271116 non-null  object
 2   Gender  271116 non-null  int64 
 3   Age     271116 non-null  int64 
 4   Height  271116 non-null  int64 
 5   Weight  271116 non-null  int64 
 6   Team    271116 non-null  object
 7   region  271116 non-null  object
 8   Year    271116 non-null  int64 
 9   Season  271116 non-null  int64 
 10  City    271116 non-null  object
 11  Sport   271116 non-null  object
 12  Event   271116 non-null  object
 13  Medal   271116 non-null  int64 
dtypes: int64(8), object(6)
memory usage: 29.0+ MB


* more cleaner data than before
* if needed we can even change the rest of the obj value to int so that we can even do analysis with that too